### ISSUE TITLE
Search case insensitive

### ISSUE THREAD

**Issue by harrymt** [christian-fei/Simple-Jekyll-Search#93]
When I perform a Fuzzy search, I want that searching string to be converted to lower case. So searching ‘CSS’ will match a tag for ‘css’. 

Can I convert the search string to lower case? If so, how?

**Comment by harrymt** [comment]
The function in question is:

https://github.com/christian-fei/Simple-Jekyll-Search/blob/2eb956578ee9e0f69448a04c09e8e1868ed9b094/src/index.js#L93-L102


Possible fix:

```javascript
 function registerInput () { 
   options.searchInput.addEventListener('keyup', function (e) { 
     var key = e.which 
     if (isWhitelistedKey(key)) { 
       emptyResultsContainer() 
       var query = e.target.value
       // Could turn this into an option?
       query = query.toLowerCase();
       search(query) 
     } 
   }) 
 } 
```

**Comment by christian-fei** [comment]
Can you make a pull-request with a test-case please?

**Comment by XhmikosR** [comment]
Weird, this seems to work for me already, i.e. the search is lowercase.

**Comment by XhmikosR** [comment]
OK, so I assume you are using fuzzy search. Because for literal search the searches are already lowercase, see here:

https://github.com/christian-fei/Simple-Jekyll-Search/blob/779575fbaed6c53c8ae41e434006148baeef9551/src/SearchStrategies/LiteralSearchStrategy.js#L11

**Comment by XhmikosR** [comment]
@christian-fei: how about we just make fuzzy search ignore case too? This would just be consistent with literal search.

**Comment by christian-fei** [comment]
@XhmikosR unfortunately fuzzysearch uses `charCodeAt` to make a comparison of each character between two strings. there isn't an option for case-insensitive search.

https://github.com/bevacqua/fuzzysearch/blob/master/index.js

we could do it by ourselves with a regexp.

do you wanna try @XhmikosR ?

**Comment by jamesacklin** [comment]
This isn't an actual fix and is fraught with issues, but in the meantime, a workaround is using Jekyll's `downcase` filter and adding nodes with multiple cases to `search.json`. I know this is bad.

`"title_downcase": "{{ page.title | downcase }}"`

**Comment by christian-fei** [comment]
hey @harrymt , did you manage to get it working perhaps?

**Comment by harrymt** [comment]
@christian-fei I did get it working, with the steps pointed out to https://github.com/christian-fei/Simple-Jekyll-Search/issues/93#issuecomment-347457881

**Comment by christian-fei** [comment]
Great to hear that!

**Comment by harrymt** [comment]
Note, this is fixed in #99
